src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/usecases/references/ReferencesTest.kt

```diff
@@ -0,0 +1,108 @@
+package it.krzeminski.snakeyaml.engine.kmp.usecases.references
+
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.assertions.withClue
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.comparables.shouldBeLessThan
+import io.kotest.matchers.concurrent.suspension.shouldCompleteBetween
+import io.kotest.matchers.shouldBe
+import it.krzeminski.snakeyaml.engine.kmp.api.Dump
+import it.krzeminski.snakeyaml.engine.kmp.api.DumpSettings
+import it.krzeminski.snakeyaml.engine.kmp.api.Load
+import it.krzeminski.snakeyaml.engine.kmp.api.LoadSettings
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Duration.Companion.seconds
+import kotlin.time.measureTime
+
+class ReferencesTest : FunSpec({
+    /**
+     * Create data which is difficult to parse.
+     *
+     * @param size - size of the map, defines the complexity
+     * @return YAML to parse
+     */
+    fun createDump(size: Int): String {
+        val root: LinkedHashMap<Any?, Any?> = LinkedHashMap()
+        var s1: LinkedHashMap<Any?, Any?>?
+        var s2: LinkedHashMap<Any?, Any?>?
+        var t1: LinkedHashMap<Any?, Any?>?
+        var t2: LinkedHashMap<Any?, Any?>?
+        s1 = root
+        s2 = LinkedHashMap()
+        /*
+     * the time to parse grows very quickly SIZE -> time to parse in seconds 25 -> 1 26 -> 2 27 -> 3
+     * 28 -> 8 29 -> 13 30 -> 28 31 -> 52 32 -> 113 33 -> 245 34 -> 500
+     */
```

Comment:
nit: the indent seems to be a bit off:
```suggestion
        /*
         * the time to parse grows very quickly SIZE -> time to parse in seconds 25 -> 1 26 -> 2 27 -> 3
         * 28 -> 8 29 -> 13 30 -> 28 31 -> 52 32 -> 113 33 -> 245 34 -> 500
         */
```